doc/configuration_system.md

```diff
@@ -0,0 +1,172 @@
+|         RUNTIME         |
++-------------------------+
+       /|\          |
+        |           |
+    [1] |       [2] |
+        |          \|/
++-------------------------+
+|         MEMORY          |
++-------------------------+ _
+       /|\          |      |\
+        |           |        \
+    [3] |       [4] |         \ [5]
+        |          \|/         \
++-------------------------+  +-------------------------+
+|          DISK           |  |       CONFIG FILE       |
++-------------------------+  +-------------------------+
+
+```
+
+__RUNTIME__ represents the in-memory data structures of ProxySQL used by the threads that are handling the requests. These contains the values of the global variables used, the list of backend servers grouped into hostgroups or the list of MySQL users that can connect to the proxy. Note that operators can never modify the contents of the __RUNTIME__ configuration section directly. They always have to go through the bottom layers.
+
+__MEMORY__ represents an in-memory SQLite3 database which is exposed to the outside via a MySQL-compatible interface. Users can connect with a MySQL client to this interface and query different tables and databases. The configuration tables
+available through this interface are:
+* mysql_servers -- the list of backend servers
+* mysql_users -- the list of users and their credentials which can connect to ProxySQL. Note that ProxySQL will use these credentials to connect to the backend servers as well (TODO: check)
+* mysql_query_rules -- the list of rules for routing traffic to the different backend servers. These rules can also cause a rewrite of the query, or caching of the result
+* global_variables -- the list of global variables used throughout the proxy that can be tweaked at runtime. Examples of global variables:
+```
+mysql> select * from global_variables limit 3;
++----------------------------------+----------------+
+| variable_name                    | variable_value |
++----------------------------------+----------------+
+| mysql-connect_retries_on_failure | 5              |
+| mysql-connect_retries_delay      | 1              |
+| mysql-connect_timeout_server_max | 10000          |
++----------------------------------+----------------+
+```
+* mysql_collations -- the list of MySQL collations available for the proxy to work with. See [this](http://stackoverflow.com/questions/341273/what-does-character-set-and-collation-mean-exactly) StackOverflow answer for the difference between a collation and a charset.
+* [only available in debug builds] debug_levels -- the list of types of debug statements that ProxySQL emits together with their verbosity levels. This allows us to easily configure at runtime what kind of statements we have in the log in order to debug different problems
+
+__DISK__ and __CONFIG FILE__
+
+__DISK__ represents an on-disk SQLite3 database, with the default location at
+`/tmp/proxysql.db`. Across restarts, the in-memory configs that were not
```

Comment:
Filename is always proxysql.db , what is configurable is only the datadir that by default is $PWD . So full path is $(datadir)/proxysql.db , with default $(PWD)/proxysql.db